pkg/api/metrics.go

```diff
@@ -76,6 +77,10 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext, reqDTO dtos.MetricReq
 
 	resp, err := hs.DataService.HandleRequest(c.Req.Context(), ds, request)
 	if err != nil {
+		var requestError *cloudwatch.RequestError
```

Comment:
This will never work for external backend plugins

Reply:
Yes, as I wrote into the description, it won't be like this, I imagine an errorWrapper that's going to be global and external plugins should wrap their errors in it. Otherwise, it's still going to return `5xx` errors I guess.

Reply:
Regarding my other comment. The idea has always been that an error returned from querying a data source means that it couldn't handle the query data request at all, hence unable to return any data and why we use a 500. If the error is happening for a certain query we can return the error in dataframe for certain refID and it will result in a 400

Reply:
I don't think we should forward the status code that the Cloud returns. Imagine that AWS is down (yeah) and it's returning a 500. If we simply return that, it would be interpreted as _our_ internal error (hurting our SLO for example). I think that a neater solution would be to return a 200 if we were able to send and receive a request and encode the result in the body like a JSON:

```
  Status: 200
  Message: {"statusCode": 401, "result": "the user foo doesn't have permissions for bar"}
```

Would we be able to do something like that?